### PR TITLE
Release/0.0.2

### PR DESCRIPTION
**Pull request by dannymoulton**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>io.kestros.commons</groupId>
+    <artifactId>kestros-parent-strict</artifactId>
+    <version>0.0.2</version>
+  </parent>
+
+  <groupId>io.kestros.commons</groupId>
+  <artifactId>structured-sling-models</artifactId>
```
Comment (dannymoulton) [inline pom.xml:32]: kestros-structured-sling-models

File: src/main/java/io/kestros/commons/structuredslingmodels/BasePage.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import org.apache.sling.models.annotations.Model;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base Sling Model for any page type.  Dynamically handles jcr:content resources, to allow method
+ * calls to be consistent whether the Model was adapted from the root or jcr:content resource.
+ */
+@Model(adaptables = Resource.class)
+public class BasePage extends BaseResource {
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/BasePage.java:38]: perhaps use a better name for this

File: src/main/java/io/kestros/commons/structuredslingmodels/BaseResource.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+  /**
+   * jcr:primaryType value of the Resource.
+   *
+   * @return jcr:primaryType value of the Resource.
+   */
+  @Nonnull
+  @JsonIgnore
+  public String getJcrPrimaryType() {
+    return getProperty("jcr:primaryType", StringUtils.EMPTY);
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/BaseResource.java:176]: use constant

File: src/main/java/io/kestros/commons/structuredslingmodels/BaseResource.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+  /**
+   * sling:resourceSuperType property of the current Resource.
+   *
+   * @return SuperType of the current Resource, or an empty String.
+   */
+  @JsonIgnore
+  @Nonnull
+  public String getResourceSuperType() {
+    final String resourceSuperType = getResource().getResourceSuperType();
+    if (resourceSuperType != null) {
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/BaseResource.java:199]: String Utils.isNotEmpty

File: src/main/java/io/kestros/commons/structuredslingmodels/BaseSlingModel.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+  /**
+   * Retrieve all Validators for the current Model.
+   *
+   * @return All Validators for the current Model.
+   */
+  @JsonIgnore
+  @Nonnull
+  public List<ModelValidator> getValidators() {
+    if (getValidationProviderService() != null) {
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/BaseSlingModel.java:164]: reduce code duplication

File: src/main/java/io/kestros/commons/structuredslingmodels/annotation/Property.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface Property {
+
+  /**
+   * Description of the annotated method.
+   *
+   * @return Description of the annotated method.
+   */
+  String description();
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/annotation/Property.java:41]: combine this with configurableProperty

File: src/main/java/io/kestros/commons/structuredslingmodels/utils/FileModelUtils.java [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+      final Class<T> type) throws InvalidResourceTypeException {
+    T file = adaptTo(fileResource, type);
+    if (!file.getErrorMessages().isEmpty()) {
+      throw new InvalidResourceTypeException(fileResource.getPath(), type,
+          "Failed validation checks. " + file.getErrorMessages());
+    } else if (!file.getFileType().getReadableContentTypes().contains(file.getMimeType())) {
+      file = fileResource.getResource().adaptTo(type);
+      if (file != null) {
+        throw new InvalidResourceTypeException(fileResource.getPath(), type,
+            "File mimeType '" + file.getMimeType() + "' did not match any expected types.");
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/utils/FileModelUtils.java:65]: string.format

File: src/main/java/io/kestros/commons/structuredslingmodels/utils/FileModelUtils.java [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+   * @param <T> Model type to adapt the resource to. Must extend BaseFile.
+   * @return The Resource adapted to the specified FileType.
+   * @throws InvalidResourceTypeException Thrown when the Resource cannot be adapted to the
+   *     passed type, or has an error after adaption.
+   */
+  public static <T extends BaseFile> T adaptToFileType(final BaseResource fileResource,
+      final Class<T> type) throws InvalidResourceTypeException {
+    T file = adaptTo(fileResource, type);
+    if (!file.getErrorMessages().isEmpty()) {
+      throw new InvalidResourceTypeException(fileResource.getPath(), type,
```
Comment (dannymoulton) [inline src/main/java/io/kestros/commons/structuredslingmodels/utils/FileModelUtils.java:59]: String.format

**Review by kevinobriendotca — APPROVED** [review]

**Pull request merged**